src/Symfony/Component/Form/Extension/Core/Type/FormType.php

```diff
@@ -26,6 +26,9 @@
 use Symfony\Component\PropertyAccess\PropertyAccess;
 use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
 
+/**
+ * @extends BaseType<self>
```

Comment:
this change is wrong. The templated type for `BaseType` is the one from `FormTypeInterface`, which is the type of the data of the form. That should be `mixed` here, not `self` (the data will never be a `FormType` instance)